src/sdk/cpp/examples/echo/echo_observer.cpp

```diff
@@ -20,11 +30,11 @@ static const std::string SUBSCRIBED_COUNT_EVENT(PROVIDER_TALENT_NAME+".echoCount
 
 class EchoObserver : public Talent {
    public:
-    explicit EchoObserver(std::shared_ptr<Publisher> publisher)
-        : Talent(TALENT_NAME, publisher) {
+    EchoObserver()
+        : Talent(TALENT_NAME) {
     }
 
-    schema::rules_ptr OnGetRules() const override {
+    schema::rule_ptr OnGetRules() const override {
```

Comment:
Why is it singular? Is it not confusing if "rules" is still plural in the method name?

Reply:
A `schema::Rules` is a subclass of `schema::Rule` (the `rules_ptr`and `rule_ptr` are just type aliased shared pointers for the these types).  Unlike in the other SDKs I've made it so that `OnGetRules()` can return a single rule without necessarily wrapping it in an aggregate rule, i.e. an `AndRules` or `OrRules`. In my opinion the relationship between the Rule and Rules is documented by their ancestral relationship so having this function return a `rule_ptr` is fine. Whether `OnGetRules`should be called `OnGetRule` is a bit philosophical, I'll change it if you insist but then I think that we should change `AndRules` -> `AndRule` or maby even better change it to just `And`. The same goes for the `OrRules`.

Reply:
In general I agree it is only about consistency and if it is easy to understand. For me it still feels mixed up between singular and plural